src/main/java/software/amazon/cloudwatchlogs/emf/environment/AgentBasedEnvironment.java

```diff
@@ -51,7 +51,7 @@ public String getLogGroupName() {
             String serviceName = getName();
             // for ECS services, replace "repo:tag" format with "repo-tag" to satisfy
             // log group regex
-            serviceName = serviceName.replaceAll(":", "-");
```

Comment:
What's the context for this particular change?

Reply:
`replaceAll()` uses `java.util.regex.Pattern.compile()` which can have a significant performance cost since an instance of this class will be created for every `getLogGroupName()` call.

Since we are not creating a regular expression and are only matching for 1 character, we can simply use `replace()` which is a shorter and simpler method which does the same thing.